e2e/util/wait.go

```diff
@@ -0,0 +1,142 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	argocdv1alpha1hack "github.com/ramendr/ramen/e2e/argocd"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	ocmv1b1 "open-cluster-management.io/api/cluster/v1beta1"
+	ocmv1b2 "open-cluster-management.io/api/cluster/v1beta2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/ramendr/ramen/e2e/types"
+)
+
+func WaitForApplicationSetDelete(
+	ctx types.Context,
+	cluster types.Cluster,
+	name, namespace string,
+	deadline time.Time,
+) error {
+	applicationSetObj := &argocdv1alpha1hack.ApplicationSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+
+	return waitForResourceDelete(ctx, cluster, applicationSetObj, deadline)
+}
+
+func WaitForConfigMapDelete(
+	ctx types.Context,
+	cluster types.Cluster,
+	name, namespace string,
+	deadline time.Time,
+) error {
+	cmObj := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+
+	return waitForResourceDelete(ctx, cluster, cmObj, deadline)
+}
+
+func WaitForPlacementDelete(
+	ctx types.Context,
+	cluster types.Cluster,
+	name, namespace string,
+	deadline time.Time,
+) error {
+	placementObj := &ocmv1b1.Placement{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+
+	return waitForResourceDelete(ctx, cluster, placementObj, deadline)
+}
+
+func WaitForManagedClusterSetBindingDelete(
+	ctx types.Context,
+	cluster types.Cluster,
+	name, namespace string,
+	deadline time.Time,
+) error {
+	mcsbObj := &ocmv1b2.ManagedClusterSetBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+
+	return waitForResourceDelete(ctx, cluster, mcsbObj, deadline)
+}
+
+// waitForResourceDelete waits until a resource is deleted or deadline is reached
+func waitForResourceDelete(
+	ctx types.Context,
+	cluster types.Cluster,
+	obj client.Object,
+	deadline time.Time,
+) error {
+	log := ctx.Logger()
+	kind := getKind(obj)
+	key := k8stypes.NamespacedName{
+		Name:      obj.GetName(),
+		Namespace: obj.GetNamespace(),
```

Comment:
Any reason not to use obj.Name, obj.Namespace? same other usage of the Get... helpers.

Reply:
Since `waitForResourceDelete` uses client.Object interface not specific object types. And this interface doesn't provide fields like Name and Namespace, only the getter methods GetName() and GetNamespace(). 
When I try to use obj.Name and obj.Namespace directly, the compiler errors ex:`obj.Name undefined (type client.Object has no field or method Name)`

Reply:
could not use runtime.Object to these fields.